python/lsst/pipe/tasks/isolatedStarAssociation.py

```diff
@@ -319,14 +321,13 @@ def _make_all_star_sources(self, tract_info, source_table_ref_dict):
         poly = tract_info.outer_sky_polygon
 
         tables = []
-        for visit in source_table_ref_dict:
-            source_table_ref = source_table_ref_dict[visit]
-            df = source_table_ref.get(parameters={'columns': all_columns})
-            df.reset_index(inplace=True)
+        for visit in source_table_handle_dict:
+            source_table_ref = source_table_handle_dict[visit]
+            tbl = source_table_ref.get(parameters={'columns': all_columns})
 
-            goodSrc = self.source_selector.selectSources(df)
+            goodSrc = self.source_selector.selectSources(tbl)
 
-            table = df[persist_columns][goodSrc.selected].to_records()
+            table = tbl[persist_columns][goodSrc.selected].as_array().view(np.recarray)
```

Comment:
I'm not sure why we need the `.view(np.recarray)` here?  Or why (equivalently) `np.asarray(tbl[persist_columns][goodSrc.selected])` doesn't work.